clang/test/CodeGenSYCL/esimd-accessor-ptr-md.cpp

```diff
@@ -0,0 +1,30 @@
+// RUN: %clang_cc1 -fsycl -fsycl-explicit-simd -fsycl-is-device -triple spir64-unknown-unknown-sycldevice -emit-llvm %s -o - | FileCheck %s
+
+// This test checks proper metadata generation for accessors used in ESIMD
+// kernels:
+// 1) Proper 'kernel_arg_accessor_ptr' metadata is generated by the FE for ESIMD
+//    kernels
+// 2) Pointers originating from accessors are marked with 'buffer_t' and proper
+//    argument kind.
+
+#include "Inputs/sycl.hpp"
+
+void test(int val) {
+  cl::sycl::accessor<int, 1, cl::sycl::access::mode::read_write> accessorA;
+  cl::sycl::accessor<int, 1, cl::sycl::access::mode::read> accessorB;
+
+  cl::sycl::kernel_single_task<class esimd_kernel>(
+      [=]() __attribute__((sycl_explicit_simd)) {
+        accessorA.use(val);
+        accessorB.use();
+      });
+  // --- Name
+  // CHECK: define spir_kernel void @_ZTSZ4testiE12esimd_kernel(
+  // --- Signature
+  // CHECK:   i32 addrspace(1)* nocapture "VCArgumentDesc"="buffer_t" "VCArgumentIOKind"="0" "VCArgumentKind"="2" %_arg_,
```

Comment:
Could you please elaborate what is `VCArgumentDesc`, `VCArgumentIOKind` and `VCArgumentKind` and how these things got here?

Reply:
This is metadata added by specific Genx pass in the Backendutil.cpp.

Reply:
Ah, ok. Thanks.